api/provider/aws/service_read.go

```diff
@@ -54,13 +77,72 @@ func (s *ServiceProvider) readService(clusterName, serviceID string) (*ecs.Servi
 	})
 
 	output, err := s.AWS.ECS.DescribeServices(input)
+	if err != nil || len(output.Services) == 0 {
```

Comment:
This reads nice, but unfortunately the `err != nil` case isn't specific enough, and this will lead to us mislabeling errors. We should only return a `ServiceDoesNotExist` if/when we catch the specific `awserr.Error` with the right code or message. 

```
if err != nil {
    if err, ok := err.(*awserr.Error); ok && err.Code == "xxx" {
        return nil, errors.Newf(...)
    }

    return err
}

if len(output.Services) == 0 {
    return nil, errors.Newf(..)
}

return output.Services[0], nil
```

Reply:
Aaah, gotcha.

✅